### PR TITLE
Added transform_depth_image_to_color_camera_custom functionality

### PR DESCRIPTION
**Pull request by SBoulanger**
Added the ability to transform IR images to color camera format using the k4a_depth_image_to_color_camera_custom function. There is still work to be done because I am pretty sure the depth image is being transformed as well. So it is redundant to only do this with the ir image. Although I am not sure the best way to return both depth and ir images from the c++ function. TODO in the future.

### CODE REVIEW

**Comment by SBoulanger** [comment]
Not sure what the codecov aren't an issue. Don't really know what that is about.

File: pyk4a/capture.py [hunk 1]
```diff
@@ -78,3 +84,11 @@ def transformed_color(self) -> Optional[np.ndarray]:
                 self.color, self.depth, self._calibration, self.thread_safe
             )
         return self._transformed_color
+
+    @property
+    def transformed_ir(self) -> Optional[np.ndarray]:
+        if self._transformed_ir is None and self.ir is not None and self.depth is not None:
+            self._transformed_ir = depth_image_to_color_camera_custom(
```
Comment (lpasselin) [inline pyk4a/capture.py:91]: If self._transformed_depth is None:
Save it since it's already processed.

Comment (lpasselin) [reply pyk4a/capture.py:91]: I mean save it since it costs nothing

File: pyk4a/pyk4a.cpp [hunk 2]
```diff
@@ -596,6 +600,80 @@ extern "C" {
...
+        PyArrayObject* np_custom_image;
+        if (K4A_RESULT_SUCCEEDED == res) {
+            res = k4a_image_to_numpy(depth_image_transformed, &np_depth_image);
+        }
+        if (K4A_RESULT_SUCCEEDED == res) {
+            res = k4a_image_to_numpy(custom_image_transformed, &np_custom_image);
+        }
+
+        if (K4A_RESULT_SUCCEEDED == res) {
+            return PyArray_Return(np_custom_image);
```
Comment (lpasselin) [inline pyk4a/pyk4a.cpp:668]: Would you mind trying something like
`return Py_BuildValue("OO", np_transformed_custom_image, np_transformed_depth_image);`

File: pyk4a/transformation.py [hunk 3]
```diff
@@ -17,6 +17,18 @@ def depth_image_to_color_camera(depth: np.ndarray, calibration: Calibration, thr
...
 
+def depth_image_to_color_camera_custom(
+    depth: np.ndarray, custom: np.ndarray, calibration: Calibration, thread_safe: bool
+) -> Optional[np.ndarray]:
+    """
+    Transforms depth image and custom image to color_image space
+    Return empty result if transformation failed
+    """
+    return k4a_module.transformation_depth_image_to_color_camera_custom(
+        calibration.transformation_handle, thread_safe, depth, custom, calibration.color_resolution,
```
Comment (lpasselin) [inline pyk4a/transformation.py:28]: Should we add option to select interpolation type? With default argument

File: pyk4a/pyk4a.cpp [hunk 4]
```diff
@@ -506,6 +507,9 @@ extern "C" {
             case K4A_IMAGE_FORMAT_COLOR_BGRA32:
                 pixel_size = (int)sizeof(uint32_t);
                 break;
+            case K4A_IMAGE_FORMAT_CUSTOM16:
+                pixel_size = static_cast<int32_t>(sizeof(int16_t));
```
Comment (lpasselin) [inline pyk4a/pyk4a.cpp:511]: I don't think we use c++ syntax in this file. We tried to keep it clang style for simplicity. Can this be a problem when using a clang compiler? Maybe we should change all syntax to c++ (malloc to new, casts, etc.)

**Review by lpasselin — CHANGES_REQUESTED** [review]

**Comment by lpasselin** [comment]
Thank you for the PR.

I added some small comments. Good work I'm sure we'll merge this soon!

> Hope the codecov failures aren't an issue. Don't really know what that is about.

Yeah we'll have to add a test to merge. It will propably be similar to the depth transformation tests.

**Comment by SBoulanger** [comment]
I saw your comments and made some changes. Check them out.

> Thank you for the PR.

No thank YOU for creating this great project.

**Comment by lpasselin** [comment]
Can you test that the transformed_depth is good?
````
original_transformed_depth = capture._transformed_depth
capture._transformed_depth = None  # reset transformed_depth.
transformed_ir = capture.transformed_ir
assert original_transformed_depth == capture.transformed_depth
````

If it works, I'll merge even if we don't have tests.

An (unecessary for now) optimization would be to skip generation of the np_depth if we already have a capture.transformed_depth.

**Comment by SBoulanger** [comment]
All is good! I did some tests and they all passed.

Yes some optimization could definitely be done. For now, the best practice is to just call for transformed_ir before transformed_depth so the depth transformation is not done twice.

**Pull request merged**